crates/astria-telemetry/src/metrics/builders.rs

```diff
@@ -194,14 +202,14 @@ impl BucketBuilder {
 /// A builder used to register individual metrics.
 ///
 /// It is constructed in [`ConfigBuilder::build`] and passed to [`Metrics::register`].
-pub struct RegisteringBuilder {
-    recorder: PrometheusRecorder,
+pub struct RegisteringBuilder<R> {
```

Comment:
I don't like that this infects all instances where metrics are being instantiated. I haven't dug deep into this code, but can we type erase this recorder so that we don't have to thread the `R` everywhere?